core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -96,6 +98,9 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
         exitProcess(KIWIX_INTERNAL_ERROR)
       }
     }
+    if (!sharedPreferenceUtil.prefIsRecentSearchMigrated) {
```

Comment:
I kind of have the feeling that once all the element are migrated we should have only on preference. I would put a reminder to do that at the end of the coding effort.